src/include/homestore/meta_service.hpp

```diff
@@ -121,7 +123,7 @@ class MetaBlkService {
      * @param cb : subsystem cb
      */
     void register_handler(meta_sub_type type, const meta_blk_found_cb_t& cb, const meta_blk_recover_comp_cb_t& comp_cb,
-                          bool do_crc = true);
+                          bool do_crc = true, std::vector< meta_sub_type > dependencies = {});
```

Comment:
Maybe
```c++
bool do_crc = true, std::optional< meta_subtype_vec_t > = std::nullopt);
```
so we don't have to instantiate a `std::vector` in the case it's not needed? The map default doesn't need to be re-assigned to another empty vector right?

Reply:
got your point here.